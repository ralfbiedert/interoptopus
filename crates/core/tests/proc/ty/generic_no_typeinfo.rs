use interoptopus::ffi_type;

#[ffi_type]
pub struct Foo<T> {
    x: T,
}

#[allow(dead_code)]
fn main() {}
