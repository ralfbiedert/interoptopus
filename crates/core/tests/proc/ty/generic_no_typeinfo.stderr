error[E0277]: the trait bound `T: TypeInfo` is not satisfied
 --> tests/proc/ty/generic_no_typeinfo.rs:5:8
  |
5 |     x: T,
  |        ^ the trait `TypeInfo` is not implemented for `T`
  |
help: consider restricting type parameter `T` with trait `TypeInfo`
  |
4 | pub struct Foo<T: interoptopus::lang::types::TypeInfo> {
  |                 +++++++++++++++++++++++++++++++++++++
